test/camelCaseSyntax.spec.js

```diff
@@ -0,0 +1,226 @@
+    });
+
+    it('should fail', () => {
+      expect(() => expect.clone()(123).toBeGreaterThan(456)).toThrow(
+        'expected 123 to be greater than 456'
+      );
+    });
+
+    it('should not support an assertion subsequently added to the original expect', () => {
+      const originalExpect = expect.clone();
+      const clonedExpect = originalExpect.clone();
+      originalExpect.addAssertion('<string> to foo', (expect, subject) =>
+        expect(subject).toEqual('foo')
+      );
+      expect(clonedExpect('foo').toFoo).notToBeAFunction();
+    });
+  });
+
+  describe('in a child', () => {
+    it('should succeed', () => {
+      expect.child()(456).toBeGreaterThan(123);
+    });
+
+    it('should fail', () => {
+      expect(() => expect.child()(123).toBeGreaterThan(456)).toThrow(
+        'expected 123 to be greater than 456'
+      );
+    });
+
+    it('should support an assertion subsequently added to the parent', () => {
+      const parentExpect = expect.clone();
+      const childExpect = parentExpect.child();
+      parentExpect.addAssertion('<string> to foo', (expect, subject) =>
+        expect(subject).toEqual('foo')
+      );
+      childExpect('foo').toFoo();
+    });
+  });
+
+  describe('with an assertion that is defined for the base of the given subject type', () => {
+    it('should succeed', () => {
+      function foo() {}
+      foo.bar = 123;
+      expect(foo).toHaveProperty('bar', 123);
+    });
+
+    it('should fail', () => {
+      function foo() {}
+      expect(() => expect(foo).toHaveProperty('bar', 123)).toThrow(
+        "expected function foo() {} to have property 'bar', 123"
+      );
+    });
+  });
+
+  describe('with an assertion that is not defined for the given subject type', () => {
+    it('should not expose a function', () => {
+      expect(expect(123).toContain).toBeUndefined();
+    });
+  });
+
+  describe('with a middle rocket assertion', () => {
+    it('should succeed', () => {
+      expect([1, 2, 3]).whenPassedAsParametersTo(Math.max).toEqual(3);
+    });
+
+    it('should fail', () => {
+      expect(() =>
+        expect([1, 2, 3]).whenPassedAsParametersTo(Math.max).toEqual(2)
+      ).toThrow('expected 3 to equal 2');
+    });
+  });
+
+  describe('with the expect.it equivalent', () => {
+    it('should succeed', () => {
+      expect({ foo: 123 }).toSatisfy({ foo: expect.toBeANumber() });
+    });
+
+    describe('and extra chaining with .and...', () => {
+      it('should succeed', () => {
+        expect({ foo: 123 }).toSatisfy({
+          foo: expect.toBeANumber().andToBeGreaterThan(100),
+        });
+      });
+
+      it('should fail', () => {
+        expect(() =>
+          expect({ foo: 123 }).toSatisfy({
+            foo: expect.toBeANumber().andToBeGreaterThan(200),
+          })
+        ).toThrow(
+          'expected { foo: 123 } to satisfy\n' +
+            '{\n' +
+            "  foo: expect.it('to be a number')\n" +
```

Comment:
As we proceed to making this user-ready we probably need to consider what to do here - perhaps we could switch out the expect.it type such that this would render the camel cases names? Wondering if that can fit with my comment about about keeping the .it. above as well.

Reply:
Yeah, absolutely, that would be fairly easy. We can try to make the rendering respect the syntax that was actually used in the source code, or just make an unconditional switch to camel case.